File: main/java/com/example/recipesandcookingtips/MainActivity.java
package com.example.recipesandcookingtips;


import androidx.appcompat.app.AppCompatActivity;
import android.app.AlertDialog;
import android.content.Intent;
import android.content.res.Resources;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ListView
import java.util.ArrayList;



public class MainActivity extends AppCompatActivity {
Button btn;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        btn = findViewById(R.id.btn1)

                btn.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View view) {
                        Intent intent = new Intent();
                        intent.setClass(MainActivity.this,PageActivity.class);
                        startActivity(intent);
                        finish();
                    }
                });
    }

    import android.content.res.Resources;
    import android.os.Bundle;
    import android.widget.ListView;
    import androidx.appcompat.app.AppCompatActivity;
    import java.util.ArrayList;


    public class MainActivity extends AppCompatActivity {
        private ArrayList<Recipes> RecipeList;

        @Override
        protected void onCreate(Bundle savedInstanceState) {
            super.onCreate(savedInstanceState);
            setContentView(R.layout.activity_main);

            RecipeList = new ArrayList<>();

            //getting string arrays from resource
            Resources res = getResources();
            String[] allRecipes = res.getStringArray(R.array.Recipes);
            String[] allAuthors = res.getStringArray(R.array.authors);

            pupoluateRecipeList(allRecipes, allAuthors);

            //setting adapter and listview
            RecipesAdapter adapter = new RecipesAdapter(getApplicationContext(),
                    R.layout.list_item, RecipeList);
            ListView listview = findViewById(R.id.Recipes_List);
            listview.setAdapter(adapter);

        }

        public void pupoluateRecipeList(String[] Recipes, String[] authors) {
            RecipeList.add(new Recipe(Recipes[0], authors[0], R.drawable.Cheesecakeimage));
            RecipeList.add(new Recipe(Recipes[1], authors[1], R.drawable.CookiesImage));
            RecipeList.add(new Recipe(Recipes[2], authors[2], R.drawable.CrepeImage));
            RecipeList.add(new Recipe(Recipes[3], authors[3], R.drawable.DonutsImage));
            RecipeList.add(new Recipe(Recipes[4], authors[4], R.drawable.HoneycakeImage));
            RecipeList.add(new Recipe(Recipes[5], authors[5], R.drawable.MochiImage));
            RecipeList.add(new Recipe(Recipes[6], authors[6], R.drawable.PanCakeImage));
            RecipeList.add(new Recipe(Recipes[7], authors[7], R.drawable.VanillaicecreamImage));
        }

    }

File: src/main/java/com/example/recipesandcookingtips1/MainActivity.kt
package com.example.recipesandcookingtips1

import android.content.Intent
import android.content.res.Resources
import android.os.Bundle
import android.widget.Button
import android.widget.ListView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    private lateinit var btn: Button
    private lateinit var recipeList: ArrayList<Recipe>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btn = findViewById(R.id.saveUsernameButton)
        btn.setOnClickListener {
            val intent = Intent(this, PageActivity::class.java)
            startActivity(intent)
            finish()
        }

        recipeList = ArrayList()
        /*
        // Получаем массивы строк из ресурсов
        val res: Resources = resources
        val allRecipes: Array<String> = res.getStringArray(R.array.Recipes)
        val allAuthors: Array<String> = res.getStringArray(R.array.authors)

        populateRecipeList(allRecipes, allAuthors)

        // Установка адаптера и ListView
        val adapter = RecipesAdapter(applicationContext, recipeList)
        val listView: ListView = findViewById(R.id.Recipes_List)
        listView.adapter = adapter*/
    }

    private fun populateRecipeList(recipes: Array<String>, authors: Array<String>) {
        recipeList.add(Recipe(recipes[0], authors[0], R.drawable.cheesecakeimage))
        recipeList.add(Recipe(recipes[1], authors[1], R.drawable.cookiesimage))
        recipeList.add(Recipe(recipes[2], authors[2], R.drawable.crepeimage))
        recipeList.add(Recipe(recipes[3], authors[3], R.drawable.donutsimage))
        recipeList.add(Recipe(recipes[4], authors[4], R.drawable.honeycakeimage))
        recipeList.add(Recipe(recipes[5], authors[5], R.drawable.mochiimage))
        recipeList.add(Recipe(recipes[6], authors[6], R.drawable.pancakeimage))
        recipeList.add(Recipe(recipes[7], authors[7], R.drawable.vanillaicecreamimage))
    }
}